src/mesh_analysis/waku_message_log_analyzer.py

```diff
@@ -72,66 +84,172 @@ def _get_affected_node_log(self, data_file: str):
+            futures = {executor.submit(self._get_affected_node_log, data_file): data_file for data_file in data_files}
+
+            for future in as_completed(futures):
+                try:
+                    result = future.result()
+                    match result:
+                        case Ok(log_path):
+                            logger.info(f'{log_path} dumped')
+                        case Err(_):
+                            logger.warning(result.err_value)
+                except Exception as e:
+                    logger.error(f'Error retrieving logs for node {futures[future]}: {e}')
 
     def _has_issues_in_local(self) -> bool:
         waku_tracer = WakuTracer()
         waku_tracer.with_received_pattern()
         waku_tracer.with_sent_pattern()
-        reader = FileReader(self._local_folder_to_analyze_path, waku_tracer)
+        reader = FileReader(self._folder_path, waku_tracer)
         dfs = reader.read()
 
         has_issues = waku_tracer.has_message_reliability_issues('msg_hash', 'receiver_peer_id', dfs[0], dfs[1],
-                                                                self._dump_analysis_dir_path)
+                                                                self._folder_path)
 
         return has_issues
 
-    def _has_issues_in_cluster(self) -> bool:
+    def _has_issues_in_cluster_single(self) -> bool:
         waku_tracer = WakuTracer()
         waku_tracer.with_received_pattern()
         waku_tracer.with_sent_pattern()
-        reader = VictoriaReader(self._victoria_config, waku_tracer)
+        reader = VictoriaReader(self._get_victoria_config_single(), waku_tracer)
         dfs = reader.read()
 
         has_issues = waku_tracer.has_message_reliability_issues('msg_hash', 'receiver_peer_id', dfs[0], dfs[1],
-                                                                self._dump_analysis_dir_path)
+                                                                self._folder_path)
+
+        return has_issues
+
+    def _read_logs_for_node(self, node_index, victoria_config_func) -> List[pd.DataFrame]:
+        waku_tracer = WakuTracer()
+        waku_tracer.with_received_pattern()
+        waku_tracer.with_sent_pattern()
+
+        config = victoria_config_func(node_index)
+        reader = VictoriaReader(config, waku_tracer)
+        data = reader.read()
+        logger.debug(f'Nodes-{node_index} analyzed')
+
+        return data
+
+    def _read_logs_concurrently(self, n_nodes: int) -> List[pd.DataFrame]:
+        dfs = []
+        with ProcessPoolExecutor() as executor:
+            futures = {executor.submit(self._read_logs_for_node, i, self._get_victoria_config_parallel): i
+                       for i in range(n_nodes)}
+
+            i = 0
+            for future in as_completed(futures):
+                try:
+                    df = future.result()
+                    dfs.append(df)
+                    i = i + 1
+                    if i % 50 == 0:
```

Comment:
No, that would only print it once. I wanted to see how it evolves as sometimes and depending on the size of the looks it may take a while, so after a few minutes you can estimate the total ammount of time.

Reply:
Oh, yes, my bad; you're totally right. Got it.